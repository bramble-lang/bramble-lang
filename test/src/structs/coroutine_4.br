fn my_main( p: bool) {
    let c : co i64 := init my_co(MyStruct{x: 5, y: 13, z: true});
    let w : i64 := yield c;
    let z : i64 := yield c;
    printiln w;
    printiln z;
    return;
}

co my_co (x: MyStruct) -> i64 {
    printiln x.x;
    yret x.y + x.y;
    printiln x.y;
    return x.x * x.x;
}

struct MyStruct {
    x: i64,
    y: i64,
    z: bool,
}