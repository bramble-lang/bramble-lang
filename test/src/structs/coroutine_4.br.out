5
13
26
25
